File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.cpp
class Codec {
public:

    // Encodes a list of strings to a single string.
    string encode(vector<string>& strs) {
        string encoded_string = ""; 
        for(string& s : strs){
            s = s.empty() ? "\01" : s; 
            encoded_string += s + '\n'; 
        }
        encoded_string.pop_back(); 
        return encoded_string; 
    }

    // Decodes a single string to a list of strings.
    vector<string> decode(string s) {
        vector<string> res; 
        stringstream ss(s); 
        string token; 
        while(getline(ss,token,'\n')){
            token = token == "\01" ? "" : token; 
            cout << token << ' '; 
            res.push_back(token); 
        }
        return res; 
    }
};

// Your Codec object will be instantiated and called as such:
// Codec codec;
// codec.decode(codec.encode(strs));

File: 0271-encode-and-decode-strings/0271-encode-and-decode-strings.java
public class Codec {

    // Encodes a list of strings to a single string.
    public String encode(List<String> strs) {
        String res = ""; 
        for(String s : strs){
            if(s.equals("")){
                res += "??/!!/";
            } else{
                res += s + "/!!/"; 
            }
        }
        //System.out.print(res); 
        return res; 
    }

    // Decodes a single string to a list of strings.
    public List<String> decode(String s) {
        List<String> res = new ArrayList<>(); 
        String[] split = s.split("/!!/"); 
        for(String strs : split){
            if(strs.equals("??")){
                res.add(""); 
            } else{
                res.add(strs); 
            }
        }

        return res; 
    }
}

// Your Codec object will be instantiated and called as such:
// Codec codec = new Codec();
// codec.decode(codec.encode(strs));